code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -786,3 +804,621 @@ nobiliumsuppression = INFINITY
+	id = "proto_nitrate_hydrogen_response"
+
+/datum/gas_reaction/proto_nitrate_hydrogen_response/init_reqs()
+	min_requirements = list(
+		/datum/gas/proto_nitrate = MINIMUM_MOLE_COUNT,
+		/datum/gas/hydrogen = MINIMUM_MOLE_COUNT,
+	)
+
+/datum/gas_reaction/proto_nitrate_hydrogen_response/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	if(cached_gases[/datum/gas/hydrogen][MOLES] > 150)
+		var produced_amount = min(5, cached_gases[/datum/gas/hydrogen][MOLES], cached_gases[/datum/gas/proto_nitrate][MOLES])
+		if(cached_gases[/datum/gas/hydrogen][MOLES] - produced_amount <0)
+			return NO_REACTION
+		cached_gases[/datum/gas/hydrogen][MOLES] -= produced_amount
+		cached_gases[/datum/gas/proto_nitrate][MOLES] += produced_amount * 0.5
+		energy_released -= produced_amount * 2500
+		if(energy_released)
+			var/new_heat_capacity = air.heat_capacity()
+			if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+				air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+		return REACTING
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response
+	priority = 25
+	name = "Proto Nitrate Cyrion B response"
+	id = "proto_nitrate_cyrion_b_response"
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response/init_reqs()
+	min_requirements = list(
+		/datum/gas/proto_nitrate = MINIMUM_MOLE_COUNT,
+		/datum/gas/cyrion_b = MINIMUM_MOLE_COUNT,
+		"TEMP" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
+	)
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/turf/open/location = isturf(holder) ? holder : null
+	if(cached_gases[/datum/gas/cyrion_b][MOLES] > MINIMUM_MOLE_COUNT)
+		var max_power = min(5, cached_gases[/datum/gas/cyrion_b][MOLES])
+		cached_gases[/datum/gas/cyrion_b][MOLES] = 0
+		explosion(location, max_power * 0.15, max_power * 0.45, max_power * 0.65, max_power, max_power* 2)
+		return REACTING
+	return NO_REACTION
+
+/datum/gas_reaction/pluox_formation
+	priority = 2
+	name = "Pluoxium formation"
+	id = "pluox_formation"
+
+/datum/gas_reaction/pluox_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/carbon_dioxide = MINIMUM_MOLE_COUNT,
+		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
+		/datum/gas/tritium = MINIMUM_MOLE_COUNT,
+		"TEMP" = 100,
+		"MAX_TEMP" = T0C
+	)
+
+/datum/gas_reaction/pluox_formation/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var produced_amount = min(5, cached_gases[/datum/gas/carbon_dioxide][MOLES], cached_gases[/datum/gas/oxygen][MOLES])
+	if(cached_gases[/datum/gas/carbon_dioxide][MOLES] - produced_amount <0 || cached_gases[/datum/gas/oxygen][MOLES] - produced_amount * 0.5 <0 || cached_gases[/datum/gas/tritium][MOLES] - produced_amount * 0.01 <0)
```

Comment:
```suggestion
	if(cached_gases[/datum/gas/carbon_dioxide][MOLES] - produced_amount < 0 || cached_gases[/datum/gas/oxygen][MOLES] - produced_amount * 0.5 < 0 || cached_gases[/datum/gas/tritium][MOLES] - produced_amount * 0.01 < 0)
```
Totally a style thing, but <0 looks kinda odd to me. Thoughts?

Reply:
yep you right, i thought i got most of them, i'll change those later on, if you see more link em